src/components/action-bar/action-bar.types.ts

```diff
@@ -1,9 +1,16 @@
 import { MenuItem } from '../menu/menu.types';
 
-export interface ActionBarItem extends MenuItem {
-    /**
-     * Renders the button in the action bar without their labels.
-     * Does not affect the items that are overflown into the overflow menu.
-     */
-    iconOnly?: boolean;
+/**
+ * Renders the button in the action bar without their labels.
+ * Does not affect the items that are overflown into the overflow menu.
+ */
+export type ActionBarItem = ActionBarItemWithLabel | ActionBarItemWithoutLabel;
+
+interface ActionBarItemWithLabel extends MenuItem {
+    iconOnly: true;
+    icon: string;
+}
+
+interface ActionBarItemWithoutLabel extends MenuItem {
```

Comment:
This one is with label
```suggestion
interface ActionBarItemWithLabel extends MenuItem {
```

Reply:
Woops autopilot 😅